### PR TITLE
update planter tag to 0.23.0

### LINKED ISSUE #12155: Planter.sh Needs to Use 0.23.0 To Build Prow
**What should be cleaned up or changed**:
[This](https://github.com/kubernetes/test-infra/blob/master/planter/planter.sh#L47) should be upgraded to version `0.23.0`, so we can build prow.

**Provide any links for context**:
https://github.com/kubernetes/test-infra/blob/master/planter/planter.sh#L47

### PR DESCRIPTION
**Pull request by adshmh**
Update the planter tag to 0.23.0. 
Closes #12155.

### CODE REVIEW

File: planter/planter.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ set -o nounset
 # these can be overridden but otherwise default to the current stable image
 # used to build kubernetes from the master branch
 IMAGE_NAME="${IMAGE_NAME:-gcr.io/k8s-testimages/planter}"
-TAG="${TAG:-0.22.0}"
+TAG="${TAG:-0.23.0}"
```
Comment (BenTheElder) [inline planter/planter.sh]: `docker pull gcr.io/k8s-testimages/planter:0.23.0` works

**Comment by adshmh** [comment]
Looks like it was a flake with `pull-test-infra-yamllint`. Could not get the build logs from the UI, only the following link: [failed build](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/test-infra/12629/pull-test-infra-yamllint/1128360934771789824). 
If this is sufficient for logging a flaky test issue, please let me know and I will log one.

File: planter/planter.sh (hunk 1 above)
Comment (adshmh) [reply planter/planter.sh]: Thank you for the review.

**Pull request merged**